### PR TITLE
List affected objects on a batch confirmation page

### PR DESCRIPTION
**Pull request by Kolyunya**
## List affected objects on a batch confirmation page

Disclaimer: this is just a WIP prototype to give the team an idea about the feature proposition.

This PR is targeted to give users an opportunity to list objects which are going to be affected by batch actions on a batch confirmation page.

I first thought that it's possible to just override the template but it turned out that the `query.execute()` call would not return an actual list of affected objects and `$this->admin->getModelManager()->addIdentifiersToQuery()` must be called *before* in a controller. That is why the controller refactoring is also needed.

I am targeting this branch, because we can modify the PR to preserve BC here.

## Changelog

```markdown
### Added

### Changed
- Changed the batch confirmation page to list affected objects.

### Deprecated

### Removed

### Fixed

### Security
```

## To do

- [ ] Update the tests;
- [ ] Update the documentation;
- [ ] Add an upgrade note.

### CODE REVIEW

**Comment by VincentLanglet** [comment]
There can be a loooot of element affected by the batch.

Maybe we should stop displaying elements if more than 25 (or any random number).

**Comment by Kolyunya** [comment]
@VincentLanglet makes sense. Actually it would be sufficient to just refactor the controller in a way which will allow clients to fetch affected objects in an overloaded template. Because currently there's not decent way even to make a custom template with this functionality.

File: src/Resources/views/CRUD/batch_confirmation.html.twig [hunk 1]
```diff
@@ -37,6 +37,13 @@ file that was distributed with this source code.
                 {% else %}
                     {% trans with {'%count%': data.idx|length} from 'SonataAdminBundle' %}message_batch_confirmation{% endtrans %}
                 {% endif %}
+                <ul>
+                    {% for element in datagrid.getQuery().execute() %}
```
Comment (phansys) [inline src/Resources/views/CRUD/batch_confirmation.html.twig:41]: IMO, we should avoid to execute data processing from a template.

**Comment by VincentLanglet** [comment]
Hi ! 

Do you still want to work on this @Kolyunya ? :)

**Pull request closed without merging**